akka-http/src/main/scala/akka/http/scaladsl/server/directives/ParameterDirectives.scala

```diff
@@ -79,12 +87,29 @@ object ParameterDirectives extends ParameterDirectives {
   private val _parameterSeq: Directive1[immutable.Seq[(String, String)]] =
     extract(_.request.uri.query().toSeq)
 
+  trait ParamSpec {
+    type Out
+    def apply(): Directive1[Out]
+  }
+  object ParamSpec {
+    implicit def forDirective1[T, U](value: T)(implicit pdef: ParamDef[T] { type Out = Directive1[U] }): ParamSpec { type Out = U } =
+      new ParamSpec {
+        type Out = U
+        def apply(): Directive1[Out] = pdef(value)
+      }
+    implicit def forDirective0[T](value: T)(implicit pdef: ParamDef[T] { type Out = Directive0 }): ParamSpec { type Out = Unit } =
+      new ParamSpec {
+        type Out = Unit
+        def apply(): Directive1[Unit] = pdef(value).tmap(_ => Tuple1(()))
+      }
+  }
+
   sealed trait ParamMagnet {
     type Out
     def apply(): Out
   }
   object ParamMagnet {
-    implicit def apply[T](value: T)(implicit pdef: ParamDef[T]): ParamMagnet { type Out = pdef.Out } =
+    /*implicit*/ def apply[T](value: T)(implicit pdef: ParamDef[T]): ParamMagnet { type Out = pdef.Out } =
```

Comment:
Needed to disambiguate the new and the old overloads. Fortunately, that makes it binary compatible.